src/ocaml-native-1-fixes.patch

```diff
@@ -24,78 +27,70 @@ index b5ef878..77454ed 100644
-         else if file_or_exe_exists long then A long
-@@ -61,9 +62,9 @@ let mk_virtual_solvers =
+-      A (find_tool cmd)
++      A (find_tool cmd target)
+     in Command.setup_virtual_command_solver (String.uppercase cmd) solver
    end
  
  let () =
 -  mk_virtual_solvers
 -    ["ocamlc"; "ocamlopt"; "ocamldep"; "ocamldoc";
 -    "ocamlyacc"; "menhir"; "ocamllex"; "ocamlmklib"; "ocamlmktop"; "ocamlfind"]
-+  mk_virtual_solvers "@target@-" 
-+	["ocamlc"; "ocamlopt"; "ocamldep"; "ocamlmklib"; "ocamlmktop"; "ocamlfind"];
-+  mk_virtual_solvers "" ["ocamldoc"; "ocamlyacc"; "menhir"; "ocamllex"]
++  mk_virtual_solvers "@target@-"
++    ["ocamlc"; "ocamlopt"; "ocamldep"; "ocamlmklib";
++    "ocamlmktop"; "ocamlfind"];
++  mk_virtual_solvers "" ["ocamldoc"; "ocamlyacc"; "menhir";
++    "ocamllex"]
  let ocamlc = ref (V"OCAMLC")
  let ocamlopt = ref (V"OCAMLOPT")
  let ocamldep = ref (V"OCAMLDEP")
-@@ -73,7 +74,7 @@ let ocamllex = ref (V"OCAMLLEX")
- let ocamlmklib = ref (V"OCAMLMKLIB")
+@@ -94,7 +97,7 @@ let ocamlmklib = ref (V"OCAMLMKLIB")
  let ocamlmktop = ref (V"OCAMLMKTOP")
  let ocamlrun = ref N
--let ocamlfind x = S[V"OCAMLFIND"; x]
-+let ocamlfind = (V"OCAMLFIND")
+ let ocamlfind_cmd = ref (V"OCAMLFIND")
+-let ocamlfind arg = S[!ocamlfind_cmd; arg]
++let ocamlfind _ = (V"OCAMLFIND")
  let program_to_execute = ref false
  let must_clean = ref false
  let show_documentation = ref false
-@@ -261,11 +262,19 @@ let init () =
-     (* TODO: warning message when using an option such as -ocamlc *)
-     (* Note that plugins can still modify these variables After_options.
-        This design decision can easily be changed. *)
--    ocamlc := ocamlfind & A"ocamlc";
--    ocamlopt := ocamlfind & A"ocamlopt";
--    ocamldep := ocamlfind & A"ocamldep";
--    ocamldoc := ocamlfind & A"ocamldoc";
--    ocamlmktop := ocamlfind & A"ocamlmktop";
-+	List.iter (fun (option,string) ->
-+	  (match !option with
-+	  | Sh s
-+	  | A s ->
-+		Log.eprintf "Warning : Command '-%s %s' overidden by option -use-ocamlfind" string s
-+	  | _ -> ()
-+	  );
-+	  option := S[ocamlfind; A string]
-+	) [(ocamlc,"ocamlc");
-+	   (ocamlopt,"ocamlopt");
-+	   (ocamldep,"ocamldep");
-+	   (ocamldoc,"ocamldoc");
-+	   (ocamlmktop,"ocamlmktop")]
-   end;
- 
-   let reorder x y = x := !x @ (List.concat (List.rev !y)) in
+@@ -232,7 +235,7 @@ let spec = ref (
+    "-install-lib-dir", Set_string Ocamlbuild_where.libdir, "<path> Set the install library directory";
+    "-install-bin-dir", Set_string Ocamlbuild_where.bindir, "<path> Set the install binary directory";
+    "-where", Unit (fun () -> print_endline !Ocamlbuild_where.libdir; raise Exit_OK), " Display the install library directory";
+-   "-which", String (fun cmd -> print_endline (find_tool cmd); raise Exit_OK), "<command> Display path to the tool command";
++   "-which", String (fun cmd -> print_endline (find_tool cmd "FIXME"); raise Exit_OK), "<command> Display path to the tool command";
```

Comment:
Please review this line carefully. Is it OK to replace extra argument with string "FIXME"?